src/client/cli/cmd/launch.cpp

```diff
@@ -126,12 +126,30 @@ mp::ReturnCode cmd::Launch::run(mp::ArgParser* parser)
     }
 
     auto ret = request_launch(parser);
-    if (ret == ReturnCode::Ok && !petenv_name.isEmpty() && request.instance_name() == petenv_name.toStdString())
+
+    if (ret != ReturnCode::Ok)
+        return ret;
+
+    if (MP_SETTINGS.get_as<bool>(mounts_key))
     {
-        if (MP_SETTINGS.get_as<bool>(mounts_key))
-            ret = mount_home(parser);
-        else
-            cout << fmt::format("Skipping '{}' mount due to disabled mounts feature\n", mp::home_automount_dir);
+        if (request.instance_name() == petenv_name.toStdString() && !mount_routes.count(home_automount_dir))
+        {
+            try
+            {
+                mount_routes.insert({home_automount_dir, QString::fromLocal8Bit(mpu::snap_real_home_dir())});
+            }
+            catch (const SnapEnvironmentException&)
+            {
+                mount_routes.insert({home_automount_dir, QDir::toNativeSeparators(QDir::homePath())});
+            }
+        }
+
+        for (const auto& [target, source] : mount_routes)
+            ret = mount(parser, source, target);
```

Comment:
Actually I don't think this is okay. If one mount fails then the next succeeds, I don't think the final return code of the command should be one of success. What do you think, @townsend2010? Should this mount as many directories as it can then return a failure code if any of them failed?

Reply:
> Should this mount as many directories as it can then return a failure code if any of them failed?

Yes, it should mount all it can and keep track of any failed mounts and return failure if there are any.  It would then list all of the failed mounts in the cli output.